source/framefmt.c

```diff
@@ -235,16 +252,23 @@ int aws_cryptosdk_serialize_frame(
     size_t plaintext_size,
     struct aws_byte_buf *ciphertext_buf,
     const struct aws_cryptosdk_alg_properties *alg_props) {
+    AWS_PRECONDITION(aws_cryptosdk_alg_properties_is_valid(alg_props));
```

Comment:
Is this a reasonable precondition? Could this ever be called with an invalid `alg_props` struct?

Reply:
This seems reasonable, yes.

That said, for context, this is a defense in depth: no users should ever be calling this function directly.